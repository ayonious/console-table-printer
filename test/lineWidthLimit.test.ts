import { Table } from '../index';

describe('Testing column max Width', () => {
  it('Simple Line Limit', () => {
    // Create a table
    const p = new Table({
      columns: [
        { name: 'red_left_align_index', alignment: 'left', color: 'red' },
        { name: 'right_align_text', alignment: 'right', maxLen: 10 },
        { name: 'green', alignment: 'center', color: 'green' },
      ],
    });

    // add rows with color
    p.addRow(
      {
        red_left_align_index: 2,
        right_align_text: 'This row is blue',
        green: 10.212,
      },
      { color: 'blue' }
    );
    p.addRow(
      {
        red_left_align_index: 3,
        right_align_text: 'I would like some red wine please',
        green: 10.212,
      },
      { color: 'red' }
    );
    p.addRow(
      {
        red_left_align_index: 4,
        right_align_text: 'I would like some cyan wine please',
        green: 10.212,
      },
      { color: 'cyan' }
    );
    p.addRow(
      {
        red_left_align_index: 5,
        right_align_text: 'I would like some white_bold wine please',
        green: 10.212,
      },
      { color: 'white_bold' }
    );
    p.addRow(
      {
        red_left_align_index: 7,
        right_align_text: 'I would like some green gemuse please',
        green: 20.0,
      },
      { color: 'green' }
    );
    p.addRow(
      {
        red_left_align_index: 8,
        right_align_text: 'I would like some gelb bananen bitte',
        green: 100,
      },
      { color: 'yellow' }
    );

    // print
    expect(p.render()).toMatchSnapshot();
    p.printTable();
  });

  it('Max Line Limit', () => {
    // Create a table
    const p = new Table({
      columns: [
        {
          name: 'center_left_align_index',
          alignment: 'center',
          color: 'red',
          title: 'Index',
        },
        {
          name: 'right_align_text',
          alignment: 'right',
          maxLen: 10,
          title: 'maxLen:10',
        },
        {
          name: 'green_center_align',
          alignment: 'center',
          color: 'green',
          maxLen: 8,
          title: 'maxLen:8',
        },
      ],
    });

    // add rows with color
    p.addRow({
      center_left_align_index: 1,
      right_align_text: 'This row is blue',
      green_center_align: 'This row is green column',
    });

    p.addRow({
      center_left_align_index: 10,
      right_align_text: 'This row is blue but again another line',
      green_center_align:
        'This row is green column but a little longer to make life harder',
    });

    p.addRow({
      center_left_align_index: 2,
      right_align_text: 'The last one I dont know what the color is',
      green_center_align: 'Thank god this is the last one',
    });

    // print
    expect(p.render()).toMatchSnapshot();
    p.printTable();
  });

  it('Min Line Limit', () => {
    // Create a table
    const p = new Table({
      columns: [
        {
          name: 'right_align_text',
          alignment: 'right',
          minLen: 100,
          title: 'MnWid:100,align:r',
        },
        {
          name: 'green_center_align',
          alignment: 'center',
          color: 'green',
          minLen: 20,
          title: 'MnWid:20,align:c',
        },
      ],
    });

    // add rows with color
    p.addRow({
      right_align_text: 'This row is blue',
      green_center_align: 'This row is green column',
    });

    p.addRow({
      right_align_text: 'This row is blue',
      green_center_align:
        'This row is green column but a little longer to make life harder',
    });

    p.addRow({
      right_align_text: 'The last one',
      green_center_align: 'Thank god this is the last one',
    });

    // print
    expect(p.render()).toMatchSnapshot();
    p.printTable();
  });

  it('Min/Max Line Limit', () => {
    // Create a table
    const p = new Table({
      columns: [
        {
          name: 'center_left_align_index',
          alignment: 'center',
          color: 'red',
          title: 'Index',
        },
        {
          name: 'right_align_text',
          alignment: 'right',
          minLen: 20,
          title: 'MinWidth:20',
        },
        {
          name: 'green_center_align',
          alignment: 'center',
          color: 'green',
          minLen: 10,
          maxLen: 15,
          title: 'MinWidth:10,max:15',
        },
      ],
    });

    // add rows with color
    p.addRow({
      center_left_align_index: 1,
      right_align_text: 'This row is blue',
      green_center_align: 'This row is green column',
    });

    p.addRow({
      center_left_align_index: 10,
      right_align_text: 'This row is blue but again another line',
      green_center_align:
        'This row is green column but a little longer to make life harder',
    });

    p.addRow({
      center_left_align_index: 2,
      right_align_text: 'The last one I dont know what the color is',
      green_center_align: 'Thank god this is the last one',
    });

    // print
    expect(p.render()).toMatchSnapshot();
    p.printTable();
  });
});
