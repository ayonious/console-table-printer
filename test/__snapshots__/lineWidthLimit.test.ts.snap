// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing column max Width Max Line Limit 1`] = `
"┌───────┬────────────┬──────────┐
│[37m [0m[01mIndex[0m │[37m [0m[01m maxLen:10[0m │[37m [0m[01mmaxLen:8[0m │
├───────┼────────────┼──────────┤
│[37m [0m[31m  1  [0m │[37m [0m[37m  This row[0m │[37m [0m[32mThis row[0m │
│[37m [0m[31m     [0m │[37m [0m[37m   is blue[0m │[37m [0m[32mis green[0m │
│[37m [0m[31m     [0m │[37m [0m[37m          [0m │[37m [0m[32m column [0m │
│[37m [0m[31m 10  [0m │[37m [0m[37m  This row[0m │[37m [0m[32mThis row[0m │
│[37m [0m[31m     [0m │[37m [0m[37m   is blue[0m │[37m [0m[32mis green[0m │
│[37m [0m[31m     [0m │[37m [0m[37m but again[0m │[37m [0m[32m column [0m │
│[37m [0m[31m     [0m │[37m [0m[37m   another[0m │[37m [0m[32m but a  [0m │
│[37m [0m[31m     [0m │[37m [0m[37m      line[0m │[37m [0m[32m little [0m │
│[37m [0m[31m     [0m │[37m [0m[37m          [0m │[37m [0m[32m longer [0m │
│[37m [0m[31m     [0m │[37m [0m[37m          [0m │[37m [0m[32mto make [0m │
│[37m [0m[31m     [0m │[37m [0m[37m          [0m │[37m [0m[32m  life  [0m │
│[37m [0m[31m     [0m │[37m [0m[37m          [0m │[37m [0m[32m harder [0m │
│[37m [0m[31m  2  [0m │[37m [0m[37m  The last[0m │[37m [0m[32m Thank  [0m │
│[37m [0m[31m     [0m │[37m [0m[37mone I dont[0m │[37m [0m[32mgod this[0m │
│[37m [0m[31m     [0m │[37m [0m[37m know what[0m │[37m [0m[32m is the [0m │
│[37m [0m[31m     [0m │[37m [0m[37m the color[0m │[37m [0m[32mlast one[0m │
│[37m [0m[31m     [0m │[37m [0m[37m        is[0m │[37m [0m[32m        [0m │
└───────┴────────────┴──────────┘"
`;

exports[`Testing column max Width Min Line Limit 1`] = `
"┌──────────────────────────────────────────────────────────────────────────────────────────────────────┬──────────────────────────────────────────────────────────────────┐
│[37m [0m[01m                                                                                   MnWid:100,align:r[0m │[37m [0m[01m                        MnWid:20,align:c                        [0m │
├──────────────────────────────────────────────────────────────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────┤
│[37m [0m[37m                                                                                    This row is blue[0m │[37m [0m[32m                    This row is green column                    [0m │
│[37m [0m[37m                                                                                    This row is blue[0m │[37m [0m[32mThis row is green column but a little longer to make life harder[0m │
│[37m [0m[37m                                                                                        The last one[0m │[37m [0m[32m                 Thank god this is the last one                 [0m │
└──────────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────────────────────────┘"
`;

exports[`Testing column max Width Min/Max Line Limit 1`] = `
"┌───────┬────────────────────────────────────────────┬────────────────────┐
│[37m [0m[01mIndex[0m │[37m [0m[01m                               MinWidth:20[0m │[37m [0m[01mMinWidth:10,max:15[0m │
├───────┼────────────────────────────────────────────┼────────────────────┤
│[37m [0m[31m  1  [0m │[37m [0m[37m                          This row is blue[0m │[37m [0m[32mThis row is green [0m │
│[37m [0m[31m     [0m │[37m [0m[37m                                          [0m │[37m [0m[32m      column      [0m │
│[37m [0m[31m 10  [0m │[37m [0m[37m   This row is blue but again another line[0m │[37m [0m[32mThis row is green [0m │
│[37m [0m[31m     [0m │[37m [0m[37m                                          [0m │[37m [0m[32m   column but a   [0m │
│[37m [0m[31m     [0m │[37m [0m[37m                                          [0m │[37m [0m[32m little longer to [0m │
│[37m [0m[31m     [0m │[37m [0m[37m                                          [0m │[37m [0m[32m make life harder [0m │
│[37m [0m[31m  2  [0m │[37m [0m[37mThe last one I dont know what the color is[0m │[37m [0m[32mThank god this is [0m │
│[37m [0m[31m     [0m │[37m [0m[37m                                          [0m │[37m [0m[32m   the last one   [0m │
└───────┴────────────────────────────────────────────┴────────────────────┘"
`;

exports[`Testing column max Width Simple Line Limit 1`] = `
"┌──────────────────────┬──────────────────┬────────┐
│[37m [0m[01mred_left_align_index[0m │[37m [0m[01mright_align_text[0m │[37m [0m[01mgreen [0m │
├──────────────────────┼──────────────────┼────────┤
│[37m [0m[31m2                   [0m │[37m [0m[34mThis row is blue[0m │[37m [0m[32m10.212[0m │
│[37m [0m[31m3                   [0m │[37m [0m[31m    I would like[0m │[37m [0m[32m10.212[0m │
│[37m [0m[31m                    [0m │[37m [0m[31m   some red wine[0m │[37m [0m[32m      [0m │
│[37m [0m[31m                    [0m │[37m [0m[31m          please[0m │[37m [0m[32m      [0m │
│[37m [0m[31m4                   [0m │[37m [0m[36m    I would like[0m │[37m [0m[32m10.212[0m │
│[37m [0m[31m                    [0m │[37m [0m[36m  some cyan wine[0m │[37m [0m[32m      [0m │
│[37m [0m[31m                    [0m │[37m [0m[36m          please[0m │[37m [0m[32m      [0m │
│[37m [0m[31m5                   [0m │[37m [0m[01m    I would like[0m │[37m [0m[32m10.212[0m │
│[37m [0m[31m                    [0m │[37m [0m[01m some white_bold[0m │[37m [0m[32m      [0m │
│[37m [0m[31m                    [0m │[37m [0m[01m     wine please[0m │[37m [0m[32m      [0m │
│[37m [0m[31m6                   [0m │[37m [0m[38m    I would like[0m │[37m [0m[32m10.212[0m │
│[37m [0m[31m                    [0m │[37m [0m[38msome crimson sky[0m │[37m [0m[32m      [0m │
│[37m [0m[31m                    [0m │[37m [0m[38m          please[0m │[37m [0m[32m      [0m │
│[37m [0m[31m7                   [0m │[37m [0m[32m    I would like[0m │[37m [0m[32m  20  [0m │
│[37m [0m[31m                    [0m │[37m [0m[32m      some green[0m │[37m [0m[32m      [0m │
│[37m [0m[31m                    [0m │[37m [0m[32m   gemuse please[0m │[37m [0m[32m      [0m │
│[37m [0m[31m8                   [0m │[37m [0m[33m    I would like[0m │[37m [0m[32m 100  [0m │
│[37m [0m[31m                    [0m │[37m [0m[33m       some gelb[0m │[37m [0m[32m      [0m │
│[37m [0m[31m                    [0m │[37m [0m[33m   bananen bitte[0m │[37m [0m[32m      [0m │
└──────────────────────┴──────────────────┴────────┘"
`;
